controllers/datafoundationclient.go

```diff
@@ -248,3 +275,52 @@ func (r *dataFoundationClientReconciler) reconcileCSIKMSConnectionDetailsConfigM
 
 	return nil
 }
+
+func (r *dataFoundationClientReconciler) hasOCSVolumes() (bool, error) {
+	storageClassList := &storagev1.StorageClassList{}
+	if err := r.list(storageClassList); err != nil {
+		return false, fmt.Errorf("failed to list Storage Classes: %v", err)
+	}
+	ocsStorageClass := map[string]bool{}
+	rbdProvisionerName := r.offering.Namespace + ocsRBDProvisionerSuffix
+	fsProvisionerName := r.offering.Namespace + ocsCephFsProvisionerSuffix
+	for i := range storageClassList.Items {
+		storageClass := storageClassList.Items[i]
+		if storageClass.Provisioner == fsProvisionerName || storageClass.Provisioner == rbdProvisionerName {
+			ocsStorageClass[storageClass.Name] = true
+		}
+	}
+	pvList := &corev1.PersistentVolumeList{}
+	pvCount := 0
+	if err := r.UnrestrictedClient.List(r.ctx, pvList); err != nil {
+		return false, fmt.Errorf("failed to list persistent volumes: %v", err)
+	}
+	for i := range pvList.Items {
+		scName := pvList.Items[i].Spec.StorageClassName
+		if ocsStorageClass[scName] {
+			pvCount++
+		}
+	}
+	if pvCount > 0 {
+		r.Log.Info(fmt.Sprintf("Current count of PVs blocking uninstallation: %v", pvCount))
+		return true, nil
+	}
+
+	return false, nil
```

Comment:
It does not make sense to have logic/logging of messages base on the result of this func inside the func itself.
Let's have this function be `countOCSVolumes` which returns a number and handles the decision-making process outside.